src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,400 @@
+
+<CodeGroup labels={["compare.go"]} lineNumbers={[false]}>
+
+```go
+package compare
+
+type Compare struct{}
+
+func (*Compare) IsGreater(int a, b) bool {
+	return a > b
+}
+```
+
+</CodeGroup>
+
+In order to use this from k6 test scripts we need to register the module
+by adding the following:
+
+<CodeGroup labels={["compare.go"]} lineNumbers={[false]}>
+
+```go
+import "go.k6.io/k6/js/modules"
+
+func init() {
+	modules.Register("k6/x/compare", new(Compare))
+}
+```
+
+</CodeGroup>
+
+Note that all k6 extensions should have the `k6/x/` prefix and the short name
+must be unique among all extensions built in the same k6 binary.
+
+We can then build a k6 binary with this extension by running
+`xk6 build --with xk6-compare=.`. In this case `xk6-compare` is the
+Go module name passed to `go mod init`, but in a real-world scenario
+this would be a URL.
+
+Finally we can use the extension in a test script:
+
+<CodeGroup labels={["test.js"]} lineNumbers={[true]}>
+
+```go
+import compare from 'k6/x/compare';
+
+export default function () {
+  console.log(compare.isGreater(2, 1));
+}
+```
+
+</CodeGroup>
+
+And run the test with `./k6 run test.js`, which should output `INFO[0000] true`.
+
+
+#### Additional features
+
+The k6 Go-JS bridge has a few features we should highlight:
+
+- Go method names will be converted from Pascal case to Camel case when
+  accessed in JS, as in the example above: `IsGreater` becomes `isGreater`.
+
+<!-- FIXME: Markdown magic to escape backticks in Go struct tags? <pre> doesn't work... -->
+- Similarly, Go field names will be converted from Pascal case to Snake case.
+  For example, the struct field `SomeField string` is accessible in JS
+  as the `some_field` object property. This behavior is configurable with the
+  `js` struct tag, so this can be changed with `SomeField string ``js:"someField"```
+  or the field can be hidden with `js:"-"`.
+
+- Method names prefixed with `X` are interpreted as constructors in JS,
+  and will support the `new` operator.
+  For example, defining the following method on the above struct:
+
+<CodeGroup labels={["compare.go"]} lineNumbers={[false]}>
+
+```go
+func (*Compare) XComparator() *Comparator {
+	return &Comparator{}
+}
+```
+
+</CodeGroup>
+
+  Would allow creating a `Comparator` instance in JS with `new compare.Comparator()`,
+  which is a bit more idiomatic to JS.
+
+- Methods that specify `context.Context` or `*context.Context` as the first
+  argument will be passed the `Context` instance used internally in k6,
+  which has attached some useful objects for inspecting the internal execution
+  state, such as
+  [`lib.State`](https://github.com/grafana/k6/blob/v0.33.0/lib/state.go#L43)
+  or VU state, [`lib.ExecutionState`](https://github.com/grafana/k6/blob/v0.33.0/lib/execution.go#L142),
+  and the [`goja.Runtime`](https://github.com/dop251/goja/blob/705acef95ba3654f89c969d9e792ac5f49215350/runtime.go#L162) instance
+  the VU is using to execute the script.
```

Comment:
```suggestion
  and the [`goja.Runtime`](https://github.com/dop251/goja/blob/705acef95ba3654f89c969d9e792ac5f49215350/runtime.go#L162) instance ?
  the VU is using to execute the script.
```
Seems the period is not completed :thinking:

Reply:
Not sure what you mean, it looks fine on https://mdr-ci.staging.k6.io/docs/refs/pull/346/merge/misc/k6-extensions/.